core/controllers/profile.py

```diff
@@ -258,26 +258,275 @@ def put(self) -> None:
         self.render_json({'status': status})
 
 
-class PreferencesHandler(base.BaseHandler[Dict[str, str], Dict[str, str]]):
-    """Provides data for the preferences page."""
+class EmailPreferencesDict(TypedDict):
+    """Dictionary that represents Email preferences."""
+
+    can_receive_email_updates: bool
+    can_receive_editor_role_email: bool
+    can_receive_feedback_message_email: bool
+    can_receive_subscription_email: bool
+
+
+class EmailPreferencesHandlerNormalizedPayloadDict(TypedDict):
+    """Dict representation of EmailPreferencesHandler's
+    normalized_payload dictionary.
+    """
+
+    data: EmailPreferencesDict
+
+
+class EmailPreferencesHandler(
+    base.BaseHandler[
+        EmailPreferencesHandlerNormalizedPayloadDict, Dict[str, str]
+    ]
+):
+    """Handles requests for the email preferences."""
+
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'data': {
```

Comment:
@vojtechjelinek should we change `data` to `email_preferences_data` here?

Reply:
Makes sense

Reply:
done. In frontend, I had to change it to camelCase.